MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/analytics/AnalyticsTracker.java

```diff
@@ -0,0 +1,134 @@
+package com.mapbox.mapboxandroiddemo.analytics;
+
+
+import android.os.Build;
+import android.support.annotation.NonNull;
+
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.IdentifyMessage;
+import com.segment.analytics.messages.ScreenMessage;
+import com.segment.analytics.messages.TrackMessage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by langstonsmith on 4/25/17.
+ */
+
+public class AnalyticsTracker {
+
+  private static final AnalyticsTracker INSTANCE = new AnalyticsTracker();
+
+  private String openedApp = "Opened app";
+  private String clickedOnSignInButtonEventName = "Clicked on sign in button";
+  private String clickedOnCreateAccountButtonEventName = "Clicked on create account button";
+  private String clickedOnNavDrawerSectionEventName = "Clicked on nav drawer section";
+  private String clickedOnIndividualExampleEventName = "Clicked on individual example";
+  private String sectionNameMapKey = "section name";
+  private String exampleNameMapKey = "example name";
+  private String isTabletMapValue = "tablet";
+  private String isPhoneMapValue = "phone";
+
+
+  public static AnalyticsTracker get() {
```

Comment:
@tobrun What's your opinion on double checked locking for singletons (@langsmith [for reference](http://www.java67.com/2015/09/thread-safe-singleton-in-java-using-double-checked-locking-pattern.html)) ?  I know there are a few ways to create thread-safe singletons.  

@langsmith Using this method, `getInstance()` would look something like the example provided in that article:
``` java
public class Singleton {

      private static volatile Singleton _instance;

      public static Singleton getInstance() {
        if (_instance == null) {  // Single check
          synchronized (Singleton.class) {
            if (_instance == null) {  // Double check
              _instance = new Singleton();
            }
          }
        }
        return _instance;
      }
    }
```
And `Singleton` would be replaced with `AnalyticsTracker`